src/main/java/edu/cornell/kfs/tax/dataaccess/impl/Transaction1099Summary.java

```diff
@@ -198,9 +225,117 @@
         
         this.taxBoxIncludedDvCheckStubTextMaps = Collections.unmodifiableMap(tempCheckStubTextMaps);
         this.taxBoxDvCheckStubTextsAreWhitelists = Collections.unmodifiableMap(tempWhitelistFlagsMap);
+        this.taxBoxMinimumReportingAmounts = parseReportingThresholdsForTaxBoxes(parameterService, boxNumberMappings);
+        this.defaultTaxBoxMinimumReportingAmount = computeDefaultTaxBoxMinimumReportingAmount();
+        this.filerAddressFields = parseFilerAddressFields(parameterService);
+    }
+
+    private Map<Tax1099FilerAddressField, String> parseFilerAddressFields(ParameterService parameterService) {
+        Tax1099FilerAddressField[] addressFieldKeys = Tax1099FilerAddressField.values();
+        Collection<String> filerAddressFieldsFromParameter = parameterService.getParameterValuesAsString(
+                CUTaxConstants.TAX_NAMESPACE, CUTaxConstants.TAX_1099_PARM_DETAIL, Tax1099ParameterNames.FILER_ADDRESS);
+        if (filerAddressFieldsFromParameter.size() != addressFieldKeys.length) {
+            throw new IllegalStateException("The filer address parameter has " + filerAddressFieldsFromParameter.size()
+                    + " sections, but should have had " + addressFieldKeys.length);
+        }
+        String[] addressFieldValues = filerAddressFieldsFromParameter.toArray(String[]::new);
+        Map<Tax1099FilerAddressField, String> addressFields = new EnumMap<>(Tax1099FilerAddressField.class);
+        for (int i = 0; i < addressFieldKeys.length; i++) {
+            addressFields.put(addressFieldKeys[i], addressFieldValues[i]);
+        }
+        return Collections.unmodifiableMap(addressFields);
+    }
+
+    private Map<String, TaxTableField> parseBoxNumberMappings(ParameterService parameterService) {

```

Comment:
Would a better name be buildBoxNumberMappingsFromParameter?

Reply:
Ok, I'll rework this method to give it a better name, like the one you suggested.